### PR TITLE
Upgrade OTel to stop Tokio being pinned

### PR DESCRIPTION
**Pull request by itowlson**
@calebschoepp I couldn't figure out how to migrate the custom error handler so I put in as much of their "migration guide" as I could get to compile, and left the commented-out custom handler for now.  If you have a moment to take a look and figure out the correct migration, that would be awesome.  Thank you so much for offering to pitch in.

The rest of it is my best effort at understanding how the APIs have changed from 0.25 to 0.27, but this really has been a matter of "follow the compiler errors" rather than any genuine understanding.  And I have no real way of testing it.  So please do read critically and err on the side of flagging anything that isn't obviously right!  Thanks!

### CODE REVIEW

File: crates/telemetry/src/logs.rs
Comment (calebschoepp) [inline crates/telemetry/src/logs.rs:88]: All these log changes seem correct to me.

File: crates/telemetry/src/metrics.rs
Comment (calebschoepp) [inline crates/telemetry/src/metrics.rs:41]: Metrics changes also make sense to me.

File: crates/telemetry/src/traces.rs
Comment (calebschoepp) [inline crates/telemetry/src/traces.rs:37]: You also get a ✅ here to 😄

File: crates/telemetry/src/lib.rs
Comment (calebschoepp) [inline crates/telemetry/src/lib.rs:111]: I'll try to get this working

**Review by calebschoepp — COMMENTED** [review]
I ran this locally and confirmed that traces, metrics, and logs are still correctly coming through 👍.

One oddity is that I'm observing some DEBUG logs that I feel like I shouldn't be seeing:

```
2024-11-21T16:39:35.806231Z DEBUG opentelemetry_sdk:  name="PeriodicReader.ExportTriggered" Export message received.
2024-11-21T16:39:35.821313Z DEBUG opentelemetry_sdk:  name="PeriodicReader.ExportTriggered" Export message received.
2024-11-21T16:40:35.805557Z DEBUG opentelemetry_sdk:  name="PeriodicReader.ExportTriggered" Export message received.
```

Investigating ^

**Comment by calebschoepp** [comment]
This should do it @itowlson https://github.com/itowlson/spin/pull/1. Once you have that merged I think this all LGTM.

**Review by calebschoepp — APPROVED** [review]

**Review by vdice — APPROVED** [review]

**Pull request merged**
